src/XrdTpc/XrdTpcNSSSupport.hh

```diff
@@ -0,0 +1,115 @@
+/******************************************************************************/
+/*                                                                            */
+/*               X r d T p c N S S S u p p o r t . h h                        */
+/*                                                                            */
+/* (c) 2021 by the Board of Trustees of the Leland Stanford, Jr., University  */
+/*   Produced by Brian Bockelman                                              */
+/*                                                                            */
+/* This file is part of the XRootD software suite.                            */
+/*                                                                            */
+/* XRootD is free software: you can redistribute it and/or modify it under    */
+/* the terms of the GNU Lesser General Public License as published by the     */
+/* Free Software Foundation, either version 3 of the License, or (at your     */
+/* option) any later version.                                                 */
+/*                                                                            */
+/* XRootD is distributed in the hope that it will be useful, but WITHOUT      */
+/* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or      */
+/* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public       */
+/* License for more details.                                                  */
+/*                                                                            */
+/* You should have received a copy of the GNU Lesser General Public License   */
+/* along with XRootD in a file called COPYING.LESSER (LGPL license) and file  */
+/* COPYING (GPL license).  If not, see <http://www.gnu.org/licenses/>.        */
+/*                                                                            */
+/* The copyright holder's institutional names and contributor's names may not */
+/* be used to endorse or promote products derived from this software without  */
+/* specific prior written permission of the institution or contributor.       */
+/******************************************************************************/
+
+#include <atomic>
+#include <string>
+#include <memory>
+
+#include "XrdOuc/XrdOucPinLoader.hh"
+
+// Forward dec'ls.
+class XrdSysError;
+typedef void CURL;
+
+namespace TPC {
+
+/**
+ * libcurl with the NSS backend has significant memory leaks around the CA handling
+ * code.  We have discovered that the memory leaks are *smallest* when NSS is given
+ * all the CA certificates in a single file (as opposed to many files in a directory).
+ * 
+ * This class takes a traditional grid CA directory, parses its contents, and creates
+ * a single file.
+ * 
+ * Each restart of the server this temporary file is created; further, every hour a
+ * new copy of the CAs is made.
```

Comment:
This really needs to be configurable. CA's rarely change every hour and a site might just want to have a better fit for it schedule like every 12 hours.

Reply:
Why?  Do we save significant I/O by going from hourly to every 12 hours?  In my tests, loading a CA directory was ~100-200ms.

There's a cost to adding a configuration knob; I question if it's worthwhile to give configuration options that allow one to save something like 2 seconds of CPU time per day.

One hour was selected to be so low no one could complain it is too infrequent (i.e., picks up RPM updates in a reasonable amount of time) and high enough to make the cost negligible.

Reply:
I'm not going to insist. I thought it would be a nice feature to have. It can always be added later if sites really want it.

Reply:
Ok, given time limitation, I'm going to put this as an item for the future if we're asked.